subcommand/server-acl-init/command_test.go

```diff
@@ -1008,6 +1112,71 @@ func TestRun_AlreadyBootstrapped(t *testing.T) {
 	}, consulAPICalls)
 }
 
+// Test if there is a provided bootstrap we skip bootstrapping of the servers
+// and continue on to the next step.
+func TestRun_SkipBootstrapping_WhenBootstrapTokenIsProvided(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+	k8s := fake.NewSimpleClientset()
+
+	bootToken := "aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee"
+	tokenFile, fileCleanup := writeTempFile(t, bootToken)
+	defer fileCleanup()
+
+	type APICall struct {
+		Method string
+		Path   string
+	}
+	var consulAPICalls []APICall
+
+	// Start the Consul server.
+	consulServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Record all the API calls made.
+		consulAPICalls = append(consulAPICalls, APICall{
+			Method: r.Method,
+			Path:   r.URL.Path,
+		})
+		switch r.URL.Path {
+		case "/v1/agent/self":
+			fmt.Fprintln(w, `{"Config": {"Datacenter": "dc1"}}`)
+		default:
+			// Send an empty JSON response with code 200 to all calls.
+			fmt.Fprintln(w, "{}")
+		}
+	}))
+	defer consulServer.Close()
+
+	// Create the Server Pods.
+	serverURL, err := url.Parse(consulServer.URL)
+	require.NoError(err)
+
+	// Run the command.
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI:        ui,
+		clientset: k8s,
+	}
+
+	responseCode := cmd.Run([]string{
+		"-resource-prefix=" + resourcePrefix,
+		"-k8s-namespace=" + ns,
+		"-server-address=" + serverURL.Hostname(),
+		"-server-port=" + serverURL.Port(),
+		"-bootstrap-token-file=" + tokenFile,
+		"-create-client-token=false", // disable client token, so there are less calls
+	})
+	require.Equal(0, responseCode, ui.ErrorWriter.String())
+
+	// Test that the expected API calls were made.
```

Comment:
It might be good to note here which call we're making sure doesn't happen, the bootstrapping one.